### DISCUSSION TITLE
Admin session invalidated every time the dev server resets?

### DISCUSSION THREAD

**Discussion by amunrarara** [medusajs/medusa#10680] [Q&A]
Cheers,

While running the dev server via `yarn dev`, every save causes the server to hot reload. When the reload is complete, my admin user session is logged out, prompting another login. It is a total pain in the rear.

I see in https://github.com/medusajs/medusa/releases/tag/v1.7.6 that the `sessionOptions` were exposed, and I tried setting `ttl` higher, and `rolling` to true. These didn't fix the issue.

How can I change that?

**Comment by lexwitte** [comment]
Bump. That's very frustrating DX

**Comment by lexwitte** [comment]
I've found a solution. You need to specify `redisUrl` in the `projectConfig`. It's mentioned here - https://docs.medusajs.com/resources/references/medusa-config#redisurl

> This configuration specifies the connection URL to Redis to store the Medusa server's session.

**Reply by amunrarara** [reply]
Awesome discovery @nxmad . That'll require I set up a Redis instance then. I'd been postponing, but I'll get that set up some time this week. Will circle back to report my result!
